### PR TITLE
Add connection name for the rabbitmq scaler

### LINKED ISSUE #5958: Naming the keda connections in Rabbitmq
### Proposal

Hi Team,

It would be good feature to add a name for the connection established by Keda in Rabbimq to estimate the no of connections consumed by each Keda config.

### Use-Case

When Keda establishing Rabbitmq we could not name the connections , In our product multiple micro-services are connecting to Rabbitmq we can name the connections for all the micro-services so that we can estimate the no of connections utilized by the micro-services ,  In the similar way we have 100+ keda configs connecting to Rabbitmq but we could not name the connections to understand no of connections used by keda in Rabbitmq.

We observed multiple connections are named with ? but we could not understand whether these connections are used by Keda or others.

Screenshot from 2024-07-11 09-47-58

### Is this a feature you are interested in implementing yourself?

No

### Anything else?

If this feature can be implemented in different way in the existing Keda, could you please let us know.

### PR DESCRIPTION
**Pull request by robpickerill**
Adds connection name for the rabbitmq scaler.

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

Fixes #5958

### CODE REVIEW

File: pkg/scalers/rabbitmq_scaler.go [hunk 1]
```diff
@@ -41,6 +41,7 @@ const (
 	rabbitMetricType                       = "External"
 	rabbitRootVhostPath                    = "/%2F"
 	rmqTLSEnable                           = "enable"
+	connectionName                         = "keda-rabbitmq-external-scaler"
```
Comment (JorTurFer) [inline pkg/scalers/rabbitmq_scaler.go:44]: I think that just setting `keda` is better. In the future we could support setting it by users
```suggestion
	connectionName                         = "keda"
```

**Review by JorTurFer — COMMENTED** [review]
LGTM! Could you update the changelog to include this as `improvement`?

**Comment by robpickerill** [comment]
> LGTM! Could you update the changelog to include this as `improvement`?

Done, thanks!

**Review by JorTurFer — APPROVED** [review]
LGTM! :heart: Thanks for the improvement!

File: pkg/scalers/rabbitmq_scaler.go [hunk 2]
```diff
@@ -41,6 +41,7 @@ const (
 	rabbitMetricType                       = "External"
 	rabbitRootVhostPath                    = "/%2F"
 	rmqTLSEnable                           = "enable"
+	connectionName                         = "keda"
```
Comment (mauvilsa) [inline pkg/scalers/rabbitmq_scaler.go:44]: Maybe better if it would be like `keda-<namespace>-<something>`?

**Review by mauvilsa — APPROVED** [review]

File: pkg/scalers/rabbitmq_scaler.go (hunk 2 above)
Comment (JorTurFer) [reply pkg/scalers/rabbitmq_scaler.go:44]: I've checked and `ScalerConfig` already has the information about ScaledObject and namespace, so we could include it:
https://github.com/kedacore/keda/blob/0771b354cb294887f462ca227100f6836204ea2d/pkg/scalers/scalersconfig/scalersconfig.go#L30-L33

If you build the connection name during the scaler creation, you can access to the info

Comment (robpickerill) [reply pkg/scalers/rabbitmq_scaler.go:44]: I'll make this change shortly

File: pkg/scalers/rabbitmq_scaler.go [hunk 3]
```diff
@@ -715,3 +721,11 @@ func (s *rabbitMQScaler) anonymizeRabbitMQError(err error) error {
 	errorMessage := fmt.Sprintf("error inspecting rabbitMQ: %s", err)
 	return fmt.Errorf(rabbitMQAnonymizePattern.ReplaceAllString(errorMessage, "user:password@"))
 }
+
+// connectionName is used to provide a deterministic AMQP connection name when
+// connecting to RabbitMQ. As far as I can tell, there are no restrictions
+// on the connection name:
+// https://github.com/rabbitmq/rabbitmq-server/blob/main/deps/amqp_client/src/amqp_connection.erl#L185-L196
```
Comment (JorTurFer) [inline pkg/scalers/rabbitmq_scaler.go:728]: It uses `longstr` type which has no limit that affects us
```suggestion
// connectionName is used to provide a deterministic AMQP connection name when
// connecting to RabbitMQ
```

**Review by JorTurFer — APPROVED** [review]
LGTM! ❤️

**Comment by mauvilsa** [comment]
Great, thank you! Looking forward to when this is part of a release.

**Pull request merged**